packages/react/src/components/text-input/text-input.tsx

```diff
@@ -22,15 +26,73 @@ import { TooltipProps } from '../tooltip/tooltip';
 import { inputsStyle } from './styles/inputs';
 import { useAriaConditionalIds } from '../../hooks/use-aria-conditional-ids';
 import { useId } from '../../hooks/use-id';
+import { focus } from '../../utils/css-state';
 
-export const Input = styled.input<{ isMobile: boolean; }>`
-    ${({ theme, isMobile }) => inputsStyle({ theme, isMobile })}
+const StyleInput = styled.input<{ isMobile: boolean }>`
+    ${({ theme, isMobile }) => inputsStyle({ theme, isMobile, isFocusable: false })};
+    border: 0;
+    flex: 1 1 auto;
+    min-height: 100%;
+    &:focus,
+    &:disabled {
+        border: 0;
+        box-shadow: none;
+    }
+`;
+
+interface AdornmentProps {
+    isMobile: boolean;
+    hasLeftAdornment: boolean;
+    hasRightAdornment: boolean;
+}
+
+const Adornment = styled.span<AdornmentProps>`
+    align-self: center;
+    color: ${({ theme }) => (theme.component['text-input-adornment-text-color'])};
+    display: flex;
+    padding-left: ${({ hasLeftAdornment }) => (hasLeftAdornment ? 'var(--spacing-1x)' : undefined)};
+    padding-right: ${({ hasRightAdornment }) => (hasRightAdornment ? 'var(--spacing-1x)' : undefined)};
+
+    > svg {
+        height: ${({ isMobile }) => (isMobile ? '24px' : '16px')};
+        width: ${({ isMobile }) => (isMobile ? '24px' : '16px')};
+    }
+`;
```

Comment:
Je pense que j'exploserais en 3 components. Ça simplifierait le render plus bas.
```suggestion
interface AdornmentProps {
    $isMobile: boolean;
}

const Adornment = styled.span<AdornmentProps>`
    align-self: center;
    color: ${({ theme }) => theme.component['text-input-adornment-text-color']};
    display: flex;

    > svg {
        height: ${({ $isMobile }) => ($isMobile ? '24px' : '16px')};
        width: ${({ $isMobile }) => ($isMobile ? '24px' : '16px')};
    }
`;

const LeftAdornment = styled(Adornment)`
    padding-left: var(--spacing-1x);
`;

const RightAdornment = styled(Adornment)`
    padding-right: var(--spacing-1x);
`;
```

Reply:
Et corrigerait le bug aussi.

Reply:
Bonne idée, c'est une fasson de faire qui m'avait pas passé par la tête.
Merci !